### ISSUE TITLE
Option to hide Title Bar

### ISSUE THREAD

**Issue by namzo** [coteditor/CotEditor#1261]
**Is your feature request related to a problem? Please describe.**
I use CotEditor in a minimal mode quite often, so I hide all distractions. I can hide most things but I cannot hide the Title Bar, and find it distracting.

**Describe the solution you'd like**
An option to hide the window's Title Bar from the "View" menu item or "Window" tab in Preferences.

**Describe alternatives you've considered**
I know there is a way to hide the Toolbar, Navigation Bar, and Tab Bar but I would like to be able to hide the Title Bar also.

**Additional context**
Mockup:
![Hide Menubar](https://user-images.githubusercontent.com/1392665/155608332-1f3ad5cd-ad22-4149-b9c3-b6fb97322494.png)

**Comment by scoskey** [comment]
Note it is hidden in full screen mode, the least distracting of all possible modes. Sorry if that's not particularly helpful.

**Comment by p1x44r** [comment]
I'd love for this feature to be implemented as well.

I actually want to keep the toolbar, but it is currently integrated into the title bar (judging by the same horizontal space both take up) and that makes the brown bar on top on the window two to two and half times taller than most other macOS applications.

Just esthetics, I know, but I'm one of those OCD people that themes and tweaks every application to look alike.

**Comment by Obadiahcruz** [comment]
+1 for this feature! would make a great of a difference when using the opacity features along with it. 

Any updates ?

**Comment by Obadiahcruz** [comment]
something like this would be great 


https://github.com/coteditor/CotEditor/assets/22825246/2bc59ddf-1de7-4f05-a5eb-ad9e8e1878ae

**Comment by 1024jp** [comment]
Let me have time to consider for this request because CotEditor already does very hacky thing for the transparent editor and has many options for window settings.
I suppose this feature works only when all the toolbar, tab bar and navigation bar are hidden.
I need to sort out those situations.
I suppose @ocreturner's idea, auto hide title bar, is smart but this is also not easy to implement (I've done the same future in another application. So I already know how to do it).
I understand hiding the title bar is attractive for users who use CotEditor with a minimalistic layout.

**Comment by ocreturner** [comment]
truly appreciate you wanting to take the time to consider this @1024jp ! i use coteditor everyday with transparent features on the side to replicate iterm2. 

however, I currently also use sticky notes with low opacity to put on top of other apps because coteditor distracts me for quick and important notes due to the title bar

allowing this feature would definitely make me remove all other apps and just use coteditor for everything  ◡̈

**Comment by 1024jp** [comment]
I have repeatedly pondered whether this feature should be implemented, and actually once implemented, but today I have made a decision: I will not implement it.
With the various display options already available for each part of the window, I believe that this option would have to be valid in a very limited state.
Perhaps those who want this feature have already hidden those parts and there would be no conflict there.
But even though, it would make the CotEditor needlessly complicated (both in coding and for users).

Sorry for those who waited.

![Screenshot 2024-05-30 at 12 50 23](https://github.com/coteditor/CotEditor/assets/1165044/bb5fceb7-8cd8-4e92-9d5a-36e2bff7f29f)

**Comment by ocreturner** [comment]
yea, I guess it was just to paint the title bar with the same color of the editor to make it translucent as well as the background color. kinda like how the bear app and obsidians have the background style options. I understand why you would reject it though. truly beautiful app. thank you for always having considered it nonetheless.

**Comment by u35253** [comment]
I'm late to the topic here, but that "Hide document title bar" option previewed 2 comments back to result in what's seen in "something like this would be great" further up would be a very sweet option to have in my humble opinion.

If CotEditor could have no title bar (and maybe even "hide scrollbar when no scrollable content"), then I would really love that!

- I have looked high and low for a "fully chromeless" text editor for macOS (that also allows customizing the font size, color, and background color).  I've landed on CotEditor as a go-to (with PlainPad as a close second), which lack the hidden-titlebar feature.  But both still have the title bar, and I'm looking for all-one-color background *without using true fullscreen mode*!

- This would be useful for quick-n-effective "editing live title slides", without the macOS chrome getting displayed as a distraction.  It's not real bad with the title bar, but the proposed change (as annoying as is to implement) would be simply superlative in my opinion.  I don't know anything else that allows this, and CotEditor seems such a good candidate, given that it is so nice in other regards.

In conclusion. I hope you reconsider, unlikely as it may be!  And, lacking that, maybe someone can suggest a next-best option (i.e., a lightweight text editor with color options that DOES support hidden title bar, even if many of the other features CotEditor has are not there).

Anyway, much gratitude for the useful tool nonetheless.
